### PR TITLE
Scope doxygen generation to include only *.h and *.md files

### LINKED ISSUE #967: Code snippets from source files showing up in API docs
These code snippets are implementation detail and shouldn't be visible in the public API docs, since they introduce noise and shouldn't be relevant to the end user.

The code is coming from the `*.c` source files which should be ignored for docs generation.

image
https://github.com/Azure/azure-sdk-for-c/blob/dc533e1db8f030dd94161e78a155ad66134a0d8a/sdk/src/azure/core/az_span.c#L244-L255

image
https://github.com/Azure/azure-sdk-for-c/blob/dc533e1db8f030dd94161e78a155ad66134a0d8a/sdk/src/azure/core/az_log.c#L27-L30

### PR DESCRIPTION
**Pull request by danieljurek**
Fixes #967

### CODE REVIEW

**Comment by ahsonkhan** [comment]
Do you have the generated docs with this change that I can look at to make sure all the data structures and files that we expect are still there?

**Comment by danieljurek** [comment]
@ahsonkhan -- https://djurekdocs.z5.web.core.windows.net/no-snippets/index.html

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -322,12 +322,10 @@ jobs:
     # Generate Documentation
     - pwsh: |
         $setupFile="doxygen-1.8.18-setup.exe";
+        $url = "https://azuresdktooling.blob.core.windows.net/dependencies/$setupFile"
```
Comment (ahsonkhan) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:325]: nit: Can we separate this change out into its own PR and leave the scope of this PR just a targeted fix for the issue?

**Comment by ahsonkhan** [comment]
I don't think this is a concern, but just throwing it out there. Is there a way to hide the source for APIs that are in-lined, and hence their implementation is visible in the header file? I think showing them is fine, but curious to know if we have an option.

image
https://github.com/Azure/azure-sdk-for-c/blob/dc533e1db8f030dd94161e78a155ad66134a0d8a/sdk/inc/azure/core/az_span.h#L290-L293

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml (hunk 1 above)
Comment (danieljurek) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml]: Sure. This didn't improve perf as much as hoped either. `Invoke-WebRequest` can get slow and unreliable when working with large files and I've been trying to eliminate it as I come across it.

**Comment by danieljurek** [comment]
@ahsonkhan -- there is the option to eliminate all source inlining.

Have a look at the docs: https://djurekdocs.z5.web.core.windows.net/disable-inlining/index.html -- 

az_span_fill() -- https://djurekdocs.z5.web.core.windows.net/disable-inlining/az__span_8h.html#a67199badce70b5ae10f879a0b977463c

**Comment by ahsonkhan** [comment]
I am not sure what is common practice for C projects and doxygen generated docs, so maybe we would revert this if other projects show the inlined sources, but I do like the docs without seeing code-snippets, even for inlined methods.

Also, looks like it also solves the duplicate enum issue: https://github.com/Azure/azure-sdk-for-c/issues/970
https://djurekdocs.z5.web.core.windows.net/disable-inlining/az__result_8h.html#afdb1073d8e61c1412ec9cc4f5536fd32

**Comment by danieljurek** [comment]
The default value for `INLINE_SOURCES` is `NO` and I'm not seeing widespread changing of this value.
